### PR TITLE
Add dockerfile for docker support

### PR DESCRIPTION
**Pull request by Kaydax**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM rust:1.75
+WORKDIR /app
+COPY . .
+RUN cargo install --path .
+CMD ["mineginx"]
```
Comment (Folleach) [inline Dockerfile]: Hello, thank you for your pull request!

Can I ask some questions?  
So, what's about use `scratch` as base image for our mineginx container?

Due to this, the result container will become very small

```suggestion
FROM rust:1.75 as build
WORKDIR /src
COPY . .
RUN rustup target add x86_64-unknown-linux-musl
RUN cargo b -r --target=x86_64-unknown-linux-musl

FROM scratch
WORKDIR /app
COPY --from=build ./src/target/x86_64-unknown-linux-musl/release/mineginx .
COPY --from=build ./src/config/ ./config/
CMD [ "./mineginx" ]
```

That is my builded containers
first is based on `rust:1.75`
second is based on `scratch`

```
folleach:~# docker image ls
REPOSITORY        TAG       IMAGE ID       CREATED             SIZE
second            latest    a3e0db460954   9 minutes ago       5.58MB
first             latest    c6918457f70b   About an hour ago   2.87GB
``` 

In addition, I can do pipeline for image publishing to the DockerHub, if you wish

Comment (Kaydax) [reply Dockerfile]: Let me try this and use it in my deployment real quick. One thing is that it seems you need to build mineginx before building the docker image as you are just copying in the binary. Mine builds the binary and installs it as a global binary in the container

Comment (Folleach) [reply Dockerfile]: Thanks for the clarification, I'm not very good at docker.  
It seems to me that there is no difference in maintenance between an installed as a global binary in a container or a copied statically-linked binary.

Of course, I don't even have bash in my container, but I don't think need it.  

Write as you try this, if you have any more fixes that have been found in your environment - welcome

**Comment by Kaydax** [comment]
Updated the Dockerfile so now it should use FROM scratch. Thanks for telling me about this as I never even knew that existed

**Comment by Folleach** [comment]
I've published this docker image to DockerHub
https://hub.docker.com/r/folleach/mineginx

Happy deployment ^_^

**Pull request merged**